src/riscv32_mmu.c

```diff
@@ -103,60 +148,144 @@ static void tlb_put(riscv32_vm_state_t* vm, uint32_t addr, uint32_t page_addr, u
 }
 
 // Virtual memory addressing mode (SV32)
-static bool riscv32_mmu_translate_sv32(riscv32_vm_state_t* vm, uint32_t addr, uint8_t access, uint32_t* dest_addr)
+bool riscv_mmu_translate(riscv32_vm_state_t *vm, const virtaddr_t va, physaddr_t *pa, const uint8_t access, bool update_pages)
```

Comment:
The soft MMU is a lot slower now, Linux system boot takes twice as much time, ramspeed reports similar regressions in raw memory speed. I think SV32 & SV39/48 translations should be separate functions as intended previously, and more optimally written, otherwise the compiler optimizations are less efficient.